src/export/InstanceExporter.ts

```diff
@@ -184,8 +185,53 @@ export class InstanceExporter implements Fishable {
             sourceInfo: rule.sourceInfo
           });
         }
-      } catch (e) {
-        logger.error(e.message, rule.sourceInfo);
+      } catch (originalErr) {
+        // if an Instance has an id that looks like a number, bigint, or boolean,
+        // we may have tried to validate with that value instead of an Instance.
+        // try to fish up an Instance with the rule's raw value.
+        // if we find one, try validating with that instead.
+        if (
+          originalErr instanceof MismatchedTypeError &&
+          (typeof rule.value === 'number' ||
+            typeof rule.value === 'bigint' ||
+            typeof rule.value === 'boolean')
```

Comment:
Wherever you have this (I think it's at least four places), consider this instead:
```ts
['number', 'bigint', 'boolean'].includes(typeof rule.value)
```
It's not really a big deal, but `prettier` let's it all fit on one line. But this is a suggestion only. Feel free to ignore.